### DISCUSSION TITLE
Don't support too old Qt versions

### DISCUSSION THREAD

**Discussion by tsujan** [lxqt/lxqt#1974] [General]
Dear developers,

Having seen the troubles we recently had with Qt5.12 and considering code readability/simplicity, I suggest we stop supporting too old Qt versions and focus on the new methods in future versions of LXQt.

If you agree, feel free to bump Qt's minimum version to 5.15 and remove version checks in your future PRs. If Qt6 becomes ready for use, even 5.15 will be too old.

**Comment by luis-pereira** [comment]
@tsujan I do agree.

**Comment by adjamhub** [comment]
I also like a lot the idea.
I'll start pushing patches about if everyone is ok on this :)

**Reply by tsujan** [reply]
> ... if everyone is ok on this

To me, it's OK. You did a great job with signal-slot connections.

Please just skip `lxqt-powermanagement`, because of a PR that does that.

**Reply by tsujan** [reply]
EDIT: Also `libqtxdg` already has a PR for it.

**Reply by adjamhub** [reply]
ok, no problem :)

**Comment by guek** [comment]
I'm more a user only of lxqt.
But which LTS distribution(s) do you support? (l)ubuntu 20.4.2 is still on QT 5.12.8 and lxqt 0.14.1.
Qt offers (commercial-only) LTS for 5.15.
Which plans has the lxqt team for a LTS version?

Please, care about lxqt users and distribution maintainers!

**Reply by tsujan** [reply]
An LTS distro is for users who want to forget about the latest changes everywhere for more than a year. There's nothing wrong with that but, as the first comment on this page implies, we can't support it.

**Comment by guek** [comment]
It's not about "latest features", but getting things done error-free and secure.

**Reply by tsujan** [reply]
Only if we had triple the manpower we do now...

**Reply by guek** [reply]
So LXDE is dead und lxqt became - so to say - for hobbyists only.
And Lubuntu becomes soon obsolete too.
Sad to hear :-(

Best wishes to you, "The Last of the Mohiqans"!

**Reply by tsujan** [reply]
> lxqt became - so to say - for hobbyists only.

Users of non-LTS distros aren't "hobbyists".

If you can't help, don't rant at least.

**Reply by guek** [reply]
This was not meant as "rant" but determination. AFAIK users of a computer as a tool love stable usable systems. And still today Linux distro upgrades are somehow adventures.

Only (software) developers find working systems boring. ;-)

Peace!

**Reply by tsujan** [reply]
Mentioning some facts, peacefully and respectfully:

 * LTS doesn't have a single meaning. I use the latest LTS kernel with a more or less bleeding edge distro — currently 5.10.32.
 * An LTS distro is a very poor environment for a serious development.
 * No group of developers have been volunteered to maintain an LTS version of LXQt and the current devs absolutely don't have time for it.

**Reply by stefonarch** [reply]
> * No group of developers have been volunteered to maintain an LTS version of LXQt and the current devs absolutely don't have time for it.

This goes also probably for the Lubuntu team-  it is well possible to have LXQt 0.17 running on ubuntu 20.04 or on debian bullseye, compiling it. Usually people who say LTS refer to *ubuntu.

**Reply by tsujan** [reply]
> This goes also probably for the Lubuntu team

Yes, that's downstream and I've seen that some people mistake it for the upstream. We don't work for a specific distro here.

**Reply by guek** [reply]
Well once upon a time LXDE meant best using Lubuntu. So one had a "dedicated" distro. That was great.

**Reply by tsujan** [reply]
LXQt doesn't have any special relation to Lubuntu; the latter uses the former — usually an outdated version of it.

**Comment by motocyklista** [comment]
Is it mean that if I will use LXQt 0.14.1 for example from distros like Lubuntu 20.04 or other I will be not get any updates? Security updates too? Can you explain me? The decision is final ? I am beginner and I didn't compile any sofware yet but maybe this is the first time when I will be must do this. This mean that if I want to use LXQt 0.17 I should compile this by myself? Or this will be uploaded to repository for example for Ubuntu?

**Reply by tsujan** [reply]
> if I will use LXQt 0.14.1 for example from distros like Lubuntu 20.04 or other I will be not get any updates?

Correct.  LTS distros always have old versions of everything. LXQt isn't an exception.

> The decision is final?

It isn't a new decision. This page was opened for LXQt developers.

**Reply by motocyklista** [reply]
@tsujan like always your answer is very helpful and professional but I have additional question to old LXQt version. For example if I will install LTS distros like Lubuntu like we know I will not get updates for LXQt environments. It is mean that I will not get security updates too? I mean that only new version of LXQt warranted that all security vulnebralities was fixed? Or maybe you will create backports for security fix to old LXQt releases? I readed some topic and I think that not but I want to have sure.

**Reply by tsujan** [reply]
As the title says, this about *too old* Qt versions. If you have an LTS distro, your Qt will become old after a year or so (it depends on how Qt is developed). Then, no LXQt update will be possible on your distro until its next upgrade. Even before that, whether you'll receive LXQt updates will depend on how active the maintainers of your distro are.

LXQt doesn't have a security update separate from its total updates. Security updates are provided by projects that have an LTS version; LXQt doesn't (read above).

**Reply by motocyklista** [reply]
@tsujan thank you for your answer. Now I know how it is work.

**Comment by luis-pereira** [comment]
Done. Minimum Qt required version is 5.15.0

**Comment by antis81** [comment]
@tsujan Without much explanation… You say 0.17 is the last version compilable against Qt 5.12.8? It's only a question of someone building and testing the packages for 20.04 before making them available to public. I will talk to Lubuntu QA team and we shall get the ball rolling again.

**Reply by tsujan** [reply]
> You say 0.17 is the last version compilable against Qt 5.12.8?

Yes.

**Reply by antis81** [reply]
Actually why is 0.17 not part of Ubuntu 20.04? It hurts my soul seeing such bad LXQt presentation! Found a whole [article](https://fosspost.org/lubuntu-20-04-review) about it - funny it doesn't even mention one of the most obvious bugs! From end-user's viewpoint the biggest issue is: "Discover" fails completely (policykit -> fixed in 0.17)! Yes, I'm a developer - I know how to use a terminal and workaround stuff. But LXQt is meant for "normal" people isn't it?

Btw. personally use Manjaro 3+ years now (love rolling-release since and no need looking back)! There are still cases LTS "fits better" (e.g. Hardware considered "too old" by Manjaro or a dedicated Hardware where reliability is a priority).

Last not least:
Actually it is not true that LXQt is "non-LTS" - you are talking about the master branch! It's okay updating the Qt version at a certain point (when avaialble or shortly after). Fixing bugs in LTS is crucial though and it is possible. Looks like just a missing communication link between Ubuntu and LXQt developers don't you think?

**Reply by tsujan** [reply]
> Actually why is 0.17 not part of Ubuntu 20.04?

You should ask that from Ubuntu maintainers.

> Actually it is not true that LXQt is "non-LTS" 

I'm afraid it is true and, most probably, will remain true.

> Looks like just a missing communication link between Ubuntu and LXQt developers don't you think?

No! LXQt is distro-agnostic. Users of all distros are welcome to make bug reports and feature requests, as long as their reports are about LXQt, not about their distros.

**Reply by tsujan** [reply]
> It hurts my soul seeing such bad LXQt presentation! Found a whole [article](https://fosspost.org/lubuntu-20-04-review) about it

As an LXQt dev, I must remain impartial toward distros and I try to be so in all places related to LXQt. Of course, personally I'm not impartial.

@agaida — who handled most reports, tested almost every change in git, managed translations, etc. (now, these jobs are done by other members) but disappeared in 2019 — spoke in a forthright way about Ubuntu. He was also the Debian maintainer of LXQt and provided Debian users with the best LXQt experience. Don't underestimate the work of package maintainers! Sadly, Debian's LXQt is outdated now.

**Reply by antis81** [reply]
>You should ask that from Ubuntu maintainers.

Already in progress!

**Reply by Borwe** [reply]
> > You should ask that from Ubuntu maintainers.
> 
> Already in progress!

If you want to build from source on Ubuntu LTS, just get the Kdeneon ppa's and you get Qt 5.15 :smile:

**Reply by antis81** [reply]
Most definitely not! :smile:

**Comment by antis81** [comment]
@guek If you have an account you can follow the discussion [here](https://www.facebook.com/groups/lubuntu.official) - or simply wait for results!

**Reply by guek** [reply]
Thanks for the info and your effort.
I'm not at FB, but I'm following lubuntu.me's telegram channel (announce & devel).

TBH I gave up using LXQT on my (main) machine and rely on plasma/ubuntu.
But as a "family-and-friends-admin" I still maintain Lubuntu 20.4.x and even debian stretch i386(!) with LXDE using components of LXQT (0.11.1 :-)

**Reply by antis81** [reply]
@guek You are in the wrong forum mate! However just wait for the TG announcement (and use whatever distro you prefer!).
